File: Server/src/main/java/jk_5/nailed/server/command/CommandTime.java
package jk_5.nailed.server.command;

import java.util.*;

import net.minecraft.command.*;
import net.minecraft.util.*;

import jk_5.nailed.api.*;
import jk_5.nailed.api.map.Map;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandTime extends NailedCommand {

    public CommandTime() {
        super("time");
    }

    @Override
    public void processCommandWithMap(ICommandSender sender, Map map, String[] args) {
        boolean hasWorld = map.getWorld() != null;
        if(args.length > 0){
            if("set".equals(args[0])){
                if(args.length > 1){
                    int target;
                    if("day".equals(args[1])){
                        target = 6000;
                    }else if("night".equals(args[1])){
                        target = 18000;
                    }else{
                        target = CommandBase.parseIntBounded(sender, args[1], 0, 23999);
                    }
                    if(!hasWorld && args.length == 3){
                        map = NailedAPI.getMapLoader().getMap(CommandBase.parseInt(sender, args[2]));
                    }
                    if(map != null){
                        map.getWorld().setWorldTime(target);
                    }
                }
            }else{
                try{
                    int number = CommandBase.parseInt(sender, args[0]);
                    map = NailedAPI.getMapLoader().getMap(number);
                    if(map != null){
                        IChatComponent component = new ChatComponentText("Current time in " + map.getSaveFileName() + ": " + map.getWorld().getWorldTime());
                        component.getChatStyle().setColor(EnumChatFormatting.GREEN);
                        sender.addChatMessage(component);
                    }
                }catch(NumberInvalidException e){
                    //NOOP
                }
            }
        }else if(hasWorld){
            IChatComponent component = new ChatComponentText("Current time in " + map.getSaveFileName() + ": " + map.getWorld().getWorldTime());
            component.getChatStyle().setColor(EnumChatFormatting.GREEN);
            sender.addChatMessage(component);
        }
    }

    @Override
    public List<String> addAutocomplete(ICommandSender sender, String[] args) {
        if(args.length == 1){
            return getOptions(args, "set");
        }else if(args.length == 2){
            return getOptions(args, "day", "night");
        }else{
            return null;
        }
    }
}

File: Server/src/main/scala/jk_5/nailed/server/command/CommandTime.scala
package jk_5.nailed.server.command

import net.minecraft.command.{CommandBase, ICommandSender}
import jk_5.nailed.api.map.Map
import net.minecraft.util.{EnumChatFormatting, ChatComponentText}
import jk_5.nailed.util.ScalaUtils._
import jk_5.nailed.api.NailedAPI

/**
 * No description given
 *
 * @author jk-5
 */
object CommandTime extends ScalaCommand {

  val name = "time"
  val usage = "/time [set] [day|night|0-23999]"

  override def processCommandWithMap(sender: ICommandSender, map: Map, args: Array[String]) = if(args.length > 0){
    caseInsensitiveMatch(args(0)) {
      case "set" =>
        if(args.length > 1){
          map.getWorld.setWorldTime(caseInsensitiveMatch(args(1)) {
            case "day" => 6000
            case "night" => 18000
            case t => CommandBase.parseIntBounded(sender, args(1), 0, 23999)
          })
        } throw new WrongUsageException("/time [set] [day|night|0-23999]")
      case _ =>
        try{
          val number = CommandBase.parseInt(sender, args(0))
          val map = NailedAPI.getMapLoader.getMap(number)
          if(map != null){
            val msg = new ChatComponentText("Current time in " + map.getSaveFileName + ": " + map.getWorld.getWorldTime)
            msg.getChatStyle.setColor(EnumChatFormatting.GREEN)
            sender.addChatMessage(msg)
          }else throw new CommandException("Unknown Map")
        }catch{
          case e: Exception =>
        }
    }
  }else{
    val msg = new ChatComponentText(s"Current time in ${map.getSaveFileName}: ${map.getWorld.getWorldTime}")
    msg.getChatStyle.setColor(EnumChatFormatting.GREEN)
    sender.addChatMessage(msg)
  }

  override def addAutocomplete(sender: ICommandSender, args: Array[String]): List[String] =
    if(args.length == 1) getOptions(args, "set")
    else if(args.length == 2) getOptions(args, "day", "night")
    else null
}
